code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,252 @@
+
+/datum/event/micro_debris/tick()
+	if(prob(60))
+		var/debris = pickweight(debris_types)
+		var/start_side = pick(cardinal)
+		var/target_zlevel = pick(GLOB.maps_data.station_levels)
+		for(var/i in 0 to rand(1,3))
+			spawn_debris(debris, start_side, target_zlevel)
+
+/datum/event/micro_debris/proc/spawn_debris(debris, start_side, zlevel)
+	var/turf/start_turf = spaceDebrisStartLoc(start_side, zlevel)
+	var/turf/destination = spaceDebrisFinishLoc(start_side, zlevel)
+	var/obj/effect/meteor/M = new debris(start_turf)
+	M.dest = destination
+	walk_towards(M, M.dest, 1)
+	return
+
+/datum/storyevent/graveyard
+	id = "graveyard"
+	name = "Space Graveryard"
+	weight = 0.6
+	parallel = FALSE
+	event_type = /datum/event/graveyard
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/graveyard
+	startWhen = 2
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/graveyard/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/graveyard/start()
+	for(var/mob/living/carbon/human/H in GLOB.human_mob_list)
+		H.sanity.sanity_damage_modifier = H.sanity.sanity_damage_modifier * 1.5
+		H.sanity.sanity_view_damage_modifier = H.sanity.sanity_view_damage_modifier * 1.5
+
+/datum/event/graveyard/announce()
+	command_announcement.Announce("Drifting wrecks of a space station have been detected near the ship. Telecommunication systems are not responsible for any strain on the crew's psychological wellbeing.", "Space Graveyard")
+
+/datum/event/graveyard/tick()
+	if(prob(5)) //random broadcasts
+		var/message = ""
+		message += pick("They are ", "He is ", "All of them are ", "I'm ", "We are ")
+		message += pick("going to die... ", "about to turn into those spider mutants... ", "being forcefully converted into commies... ")
+		message += pick("Run while you still can.", "Help!", "Angels bless our souls...", "It's... too late.")
+
+		global_announcer.autosay(msg, "Emergency Broadcast")
+	else if(prob(5)) //predetermined broadcasts
+		var/message_list = list(
+			"Blessed Angels, guide us to safety!",
+			"Comrades, we have captured the last survivors on this wreck. Expecting extraction at-",
+			"Whoever hears this, RUN WHILE YOU STILL CAN!",
+			"We are barely managing to keep this place safe. Please, whoever recieves this signal, pick us up at-",
+			"Our food and water supplies are going to run out soon. We have money. Just help us, anyone, please...",
+			"Security is... All gone. With medical bay soon to follow. These abominations know nothing but hunger, consumed most of our crew, and yet they remain unsatiated... Do not try to help in any way. This station is a lost cause."
+		)
+
+		global_announcer.autosay(pick(message_list), "Emergency Broadcast")
+	else if(prob(1)) //sekrit stuf
+		global_announcer.autosay("Man, all those people really suck. Just don't get hit and beat everything until it goes horizontal.", "Emergency Broadcast")
```

Comment:
"Click it till it horizontal" is OOC meme, sounds off IC.

Reply:
we need the funi